src/lib/classes/engine/EngineStorage.ts

```diff
@@ -0,0 +1,162 @@
+import type IEngineStorageObject from "$lib/interfaces/IEngineStorageObject";
+import { Engine } from "./Engine";
+/**
+ * Engine storage definition.
+ * Stores an array of Engine instances, and can create, delete or search the array
+ * */
+class EngineStorage {
+	/**
+	 * Array of engines defined
+	 * */
+	engineArray: Array<Engine | undefined> = [];
+
+	/**
+	 * variable used to give unique id to engines
+	 * */
+	#engineId: number = 0;
+
+	get engineId(): number {
+		return (this.engineId = ++this.#engineId); //auto inc
```

Comment:
If you are trying to read and understand the code, this can be very difficult to understand. Just increment the private counter on line 19, then return the new value on line 20.